pkg/driver/driver.go

```diff
@@ -39,3 +48,19 @@ func (d *BaseDriver) Start(_ context.Context) (chan error, error) {
 func (d *BaseDriver) Stop(_ context.Context) error {
 	return nil
 }
+
+func (d *BaseDriver) CreateSnapshot(_ context.Context, _ string) error {
+	return fmt.Errorf("unimplemented")
+}
+
+func (d *BaseDriver) ApplySnapshot(_ context.Context, _ string) error {
+	return fmt.Errorf("unimplemented")
+}
+
+func (d *BaseDriver) DeleteSnapshot(_ context.Context, _ string) error {
+	return fmt.Errorf("unimplemented")
+}
+
+func (d *BaseDriver) ListSnapshots(_ context.Context) (string, error) {
```

Comment:
As noted above, the output is from the human monitor and not parseable by machines (but scrapable)

But since we already convert it in the command, I suppose we could do the same in the driver instead.

Reply:
```go
        out, err := snapshot.List(ctx, inst)
        if err != nil {
                return err
        }
        if quiet {
                for i, line := range strings.Split(out, "\n") {
                        // "ID", "TAG", "VM SIZE", "DATE", "VM CLOCK", "ICOUNT"
                        fields := strings.Fields(line)
                        if i == 0 || line == "" {
                                // skip header and empty line after using split
                                continue
                        }
                        tag := fields[1]
                        fmt.Printf("%s\n", tag)
                }
                return nil
        }
        fmt.Print(out)

```

Reply:
Yes exactly, we could scrap it to a know object in driver itself. 
So the commands will become independent of the driver output

Reply:
Rebased with `string` (including HEADER line), needs to return an array of... something ?

The only required field in it is Tag, since that (string) is required by the other commands.

Reply:
I was referring to more of creating a struct itself with the following,
```go
type SnapshotInfo struct {
id string
tag string
vmSize string
date string
vmClock string
}
```
As part of QEMU driver we will do the exact same like we done to fetch tag, but we will do for all needed fields. After that we can return SnapshotInfo[] as part of the command.

Reply:
There really is no consistent output to parse, it is intended to be read by human.

Output while running:

```
INFO[0000] Sending HMP [info snapshots] command         
There is no snapshot available.

INFO[0000] Sending HMP [info snapshots] command         
List of snapshots present on all disks:
ID        TAG                     VM SIZE                DATE       VM CLOCK
--        snap1                  2.33 GiB 2023-02-26 18:16:36   00:06:18.553
```

Output while stopped:

```
DEBU[0000] Running qemu-img [snapshot -l /home/anders/.lima/default/diffdisk] command 

DEBU[0000] Running qemu-img [snapshot -l /home/anders/.lima/default/diffdisk] command 
ID        TAG                     VM SIZE                DATE       VM CLOCK
1         snap1                  2.33 GiB 2023-02-26 18:16:36   00:06:18.553
```

Will need to filter out the header bits, but it still seems a bit error prone.

Reply:
Sure. Thanks. We can ignore this in that case